internal/provider/provider.go

```diff
@@ -96,24 +102,32 @@ func (p *IncusProvider) Schema(_ context.Context, _ provider.SchemaRequest, resp
 					Attributes: map[string]schema.Attribute{
 						"name": schema.StringAttribute{
 							Required:    true,
-							Description: "Name of the Incus remote. Required when incus_scheme set to https, to enable locating server certificate.",
+							Description: "Name of the Incus remote. Required when incus_address scheme set to https, to enable locating server certificate.",
 						},
 
 						"address": schema.StringAttribute{
 							Optional:    true,
-							Description: "The FQDN or IP where the Incus daemon can be contacted. (default = \"\" (read from lxc config))",
+							Description: "The URL where the Incus daemon can be contacted. (default = \"\" (read from lxc config))",
+						},
+
+						"protocol": schema.StringAttribute{
+							Optional:    true,
+							Description: "Server protocol (incus, oci or simplestreams)",
+							Validators: []validator.String{
+								stringvalidator.OneOf("incus", "oci", "simplestreams"),
+							},
 						},
 
-						"port": schema.StringAttribute{
+						"public": schema.BoolAttribute{
 							Optional:    true,
-							Description: "Port Incus Daemon API is listening on. (default = 8443)",
+							Description: "Public image server",
 						},
 
-						"scheme": schema.StringAttribute{
+						"auth_type": schema.StringAttribute{
 							Optional:    true,
-							Description: "Unix (unix) or HTTPs (https). (default = unix)",
+							Description: "Server authentication type, tls or oidc. ( Only for the `incus` protocol )",
```

Comment:
In the case `auth_type` is only required for the `incus` protocol, then you should add the validation that ensures that.